Sources/AWSSDKSwiftCore/Signers/V4.swift

```diff
@@ -47,11 +47,15 @@ extension Signers {
 
         /// If you did not provide credentials `manageCredential()` should be called and the future resolved prior to building signedURL or signedHeaders to ensure latest credentials are retreived and set
         public func manageCredential() -> Future<CredentialProvider> {
-            if credential.isEmpty() || credential.nearExpiration() {
+            if credential.nearExpiration() {
                 do {
-                    return try MetaDataService.getCredential().map { credential in
-                        self.credential = credential
-                        return credential
+                    return try MetaDataService.getCredential()
+                        .flatMapError { _ in
```

Comment:
Just trying to figure out this addition. I think this means not setting access key or secret at all will lead to skip signing if the metadata service fails. Is this what we want? Im not sure it is. I’d rather surface the error so the client knows it tried to get creds from the metadata service. But if it is why not set the credential instance variable on this error handler as well so nearExpiration is always false moving fwd. then we wont try the metadata service again the next time?

Reply:
The reason for this change is there are some aws services that don’t require credentials (eg CognitoIdentityProvider) but we are requiring credentials all the time. This change is to stop the metadata service request if you explicitly set the credentials to empty. 

I’m actually wondering if we should disable the metadata service calls completely on non Linux builds because they don’t make sense outside of ECS/EC2 instances.